tools/checkdocs/checkdocs.go

```diff
@@ -220,3 +236,76 @@ func verifyTags(fm []byte) error {
 
 	return nil
 }
+
+// checkTableFile verifies that supported-commands.md is correctly formatted,
+// using logf for progress reporting and fatalf for errors.
+func checkTableFile(file string, logf, fatalf func(string, ...any)) {
+	fileInBytes, err := os.ReadFile(file)
+	if err != nil {
+		fatalf("Couldn't read file %s: %s", file, err)
```

Comment:
I think it is better to make those functions return an error, like normal Go functions